### DISCUSSION TITLE
Are Desktop Nvidia GPUs supported?

### DISCUSSION THREAD

**Discussion by codecowboy** [hollowaykeanho/Upscaler#172] [General]
The readme mentions that the project targets laptops. Can this project also make use of desktop grade NVIDIA GPUs and cuda / cudnn?

Specifically,  I would be grateful if someone could let me know whether this project will work with Ubuntu 22 or 24 and an Nvidia RTX 3090.

**Comment by jamie-belanger** [comment]
fwiw I just found this project this morning and have been playing around for the past hour or so.

There's nothing preventing this program from running on desktop hardware. I have no idea if it's _using_ the hardware to the fullest, but it does use the hardware. It does upscale images. I'm currently trying to upscale a 30 second test video right now, and it's chugging along. It feels kind of slow, but I was expecting that.

My machine:
* Linux Mint 21.3 (based on Ubuntu 22)
* Kernel: 6.8.0-58-generic 
* CPU: AMD Ryzen 5 2600X (12) @ 3.600GHz 
* GPU: NVIDIA GeForce GTX 1650 (a low to mid range card released about a year before your RTX 3090)
* RAM: 64 GB

The only issue I've had is that their instructions are incorrect. All of the examples include `--model ultrasharp` but that doesn't work. You have to use the full ID of the model, like `--model upscayl-ultrasharp`

You can see the full list of included models with `./start.cmd --help`
